portfolio/src/main/java/com/google/sps/data/GenreRecommender.java

```diff
@@ -0,0 +1,78 @@
+package com.google.sps.data;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.google.sps.data.Book;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class GenreRecommender {
+  private List<Book> books;
+  private Map<Book, Set<String>> bookToGenres;
+  private Map<String, Set<Book>> genreToBooks;
+
+  /**
+   * Constructor that takes in a list of books
+   * and populates class variables appropriately.
+   *
+   * @param books: List of Book objects for the recommendation algorithm
+   */
+  public GenreRecommender(List<Book> books) {
+    this.books = books;
+    bookToGenres = new HashMap<Book, Set<String>>();
+    genreToBooks = new HashMap<String, Set<Book>>();
```

Comment:
nit: (mostly fyi, change is optional) These can be replaced with `ImmutableSetMultiMap1`. It doesn't require you to handle the case whether key already exists in the map or not. It also returns ImmutableSet on get, so you don't have to copy it.

https://guava.dev/releases/19.0/api/docs/com/google/common/collect/ImmutableSetMultimap.html